homeassistant/components/history/__init__.py

```diff
@@ -64,6 +65,7 @@ def get_significant_states(
         query = session.query(States).filter(
             (
                 States.domain.in_(SIGNIFICANT_DOMAINS)
+                | (not significant_changes_only and States.domain.is_("device_tracker"))
```

Comment:
Don't do this in SQL. This means the database is calculating it. You shouldn't apply the filter at all.

Reply:
Done